### PR TITLE
Bumping amazonka and cleaning up

### PR DESCRIPTION
**Pull request by nhibberd**

### CODE REVIEW

File: mismi-core/src/Mismi/Control.hs [hunk 1]
```diff
@@ -75,3 +85,51 @@ awsBracket_ r f a =
...
+
+serviceRetryAWSAction :: Retry -> AWS a -> AWS a
+serviceRetryAWSAction r =
+  local (override (serviceRetry .~ r))
+
+retryAWSAction :: Int -> AWS a -> AWS a
+retryAWSAction =
+  local . retryAWS
+
+retryAWS :: Int -> Env -> Env
```
Comment (charleso) [inline mismi-core/src/Mismi/Control.hs]: I'm assuming we should set this with a sensible default in `runAWS`?

File: mismi-core/src/Mismi/Control.hs [hunk 2]
```diff
@@ -31,6 +39,9 @@ import           Network.AWS.Data as A
 import           Network.AWS.Error as A
 import           Network.AWS.Waiter as A
 
+import           Network.HTTP.Client (HttpException (..))
+import           Network.HTTP.Types.Status -- (status500, status404)
```
Comment (charleso) [inline mismi-core/src/Mismi/Control.hs]: /cruft?

**Comment by charleso** [comment]
:+1: Once the builds pass

**Comment by markhibberd** [comment]
:100:

**Comment by charleso** [comment]
:+1:

**Pull request merged**